src/ObjectUtils/DwarfUtils.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "ObjectUtils/DwarfUtils.h"
+
+#include <absl/strings/str_cat.h>
+
+#include "OrbitBase/Logging.h"
+
+namespace orbit_object_utils {
+
+namespace {
+// This function converts a type `DWARFDie` node into a string representation. Currently, it only
+// handles types that may occur as a type attribute of a DW_TAG_formal_parameter and NOT all
+// possible type Dies from the specification (https://dwarfstd.org/doc/DWARF5.pdf).
+std::string DwarfTypeDieToString(const llvm::DWARFDie& type_die) {
```

Comment:
I also think this should have some unit tests. And for this you probably wanna make this function part of the API to make it independently testable.